Docs/Architecture.md

```diff
@@ -0,0 +1,1429 @@
+
+## Overview
+
+OneCrypto is a refactoring of the MuCryptoRelease, MuSharedCrypto, and CryptoBinary concepts. This effort is focused on simplifying and improving industry adoption, reliability, and compatibility. The goal is to help all parties shipping products with cryptographic support do it better and stay up to date more easily.
+
+### Repositories
+
+- **`MuCryptoRelease`** - Repository and project for building the crypto binary and carrying crypto library dependencies like OpenSSL
+- **`UEFI-MuBaseCore-Edk2`** - Generic representation of the "fork" of common code that is common in UEFI industry, providing the interface for crypto and other common services
+
+### Key Benefits
+
+1. **Simplified Updates**: Make it much easier to update crypto for many platforms and stay current with minimal validation
+2. **Reduced Build Cost**: No need to download and compile OpenSSL for every platform build
+3. **Flash Size Savings**: One copy of crypto instead of 2-3 per phase (DXE, SMM, MM)
+4. **Better Tracking**: Unified crypto inventory for SBOM and vulnerability management
+5. **Crypto Agility**: Easy to track and update crypto usage across ecosystem
+6. **Supply Chain Benefits**: Reduced SDL tool triggers from OpenSSL test dependencies
+7. **Future Ready**: Support for Post Quantum Computing (PQC) and other algorithm changes
+
+## Code Layout
+
+The OneCrypto architecture is implemented across multiple packages, each with specific responsibilities.
+
+### MU_BASECORE
+
+The MU_BASECORE repository contains the core UEFI/EDK2 code with OneCrypto integration points. Changes here define the public interfaces that platforms and modules use.
+
+**Key Components:**
+
+- **`CryptoPkg/Include/Library/BaseCryptLib.h`**
+  - Defines the standard cryptographic API interface used by all UEFI modules
+  - This is the stable API that modules depend on
+  - No knowledge of OneCrypto implementation details
+
+- **`CryptoPkg/Include/Protocol/OneCrypto.h`**
+  - Defines the `ONE_CRYPTO_PROTOCOL` structure with versioned crypto function pointers
+  - Contains protocol GUIDs: `gOneCryptoDxeProtocolGuid`, `gOneCryptoMmProtocolGuid`, `gOneCryptoPrivateProtocolGuid`
+  - Version fields: `UINT16 Major`, `UINT16 Minor`
+  - All crypto function signatures and documentation
+
+- **`CryptoPkg/Library/BaseCryptLibOnOneCrypto/`**
+  - Implementation of BaseCryptLib that locates and calls OneCrypto protocol
+  - Phase-specific instances:
+    - `BaseCryptLibOnOneCrypto_Dxe` - Locates DXE protocol via Boot Services
+    - `BaseCryptLibOnOneCrypto_Mm` - Locates MM protocol via MM Services
+  - Checks protocol version before calling functions to ensure compatibility
+  - Thin wrapper layer with minimal logic
+
+- **`CryptoPkg/Library/BaseCryptLibNull/`**
+  - Null implementation that returns errors for all functions
+  - Used when crypto is not available or during early boot phases
+
+### CryptoBinPkg
```

Comment:
From what you've mentioned before, `CryptoBinPkg` is planned to be dropped so it should be removed from here.